### PR TITLE
Convert styeguide to ES6

### PR DESCRIPTION
**Pull request by joeybaker**
This is mostly just a port of https://github.com/airbnb/javascript/pull/276/files

I'm okay with the rules here, but please suggest modifications!
- @techwraith
  - [x] for
  - [ ] against
  - [ ] abstain
- @ben-ng 
  - [ ] for
  - [ ] against
  - [x] abstain
- @cstumph 
  - [ ] for
  - [ ] against
  - [x] abstain
- @CristiScheye 
  - [x] for
  - [ ] against
  - [ ] abstain
- @ghostbar 
  - [ ] for
  - [ ] against
  - [ ] abstain
- @joeybaker
  - [x] for
  - [ ] against
  - [ ] abstain
- @tedbreen 
  - [x] for
  - [ ] against
  - [ ] abstain
- @jmcelroy5 
  - [x] for
  - [ ] against
  - [ ] abstain
- @Jon-Biz 
  - [x] for
  - [ ] against
  - [ ] abstain

### CODE REVIEW

**Comment by ben-ng** [comment]
you need to give us access to this repo for us to sign off

**Comment by joeybaker** [comment]
!? that's weird…

**Comment by joeybaker** [comment]
Should be fixed now!

File: README.md [hunk 1]
```diff
@@ -482,145 +687,164 @@ function onThingHappended(cb){
...
-    var name = 'Bob Parr'
+    const name = 'Bob Parr'
 
     // bad
-    var fullName = "Bob " + this.lastName
+    const fullName = "Bob " + this.lastName
 
     // good
-    var fullName = 'Bob ' + this.lastName
+    const fullName = 'Bob ' + this.lastName
```
Comment (cscheye) [inline README.md]: Should we favor string interpolation here? (templates)

Comment (joeybaker) [inline README.md]: Yup!

File: README.md [hunk 2]
```diff
@@ -482,145 +687,164 @@ function onThingHappended(cb){
...
     // immediately-invoked function expression (IIFE)
-    (function(){
-      console.log('Welcome to the Internet. Please follow me.')
+    (() => {
+      console.log('Welcome to the Internet. Please follow me.');
     })()
     ```
 
-  - Never declare a function in a non-function block (if, while, etc). Assign the function to a variable instead. Browsers will allow you to do it, but they all interpret it differently, which is bad news bears.
+  - Never declare a function in a non-function block (if, while, etc). Assign the function to a variable instead. Browsers will allow you to do it, but they all interpret it differently, which is bad news bears. It's also slower.
```
Comment (cscheye) [inline README.md]: Also, aren't variables declared with `const` and `let` block scoped? So they won't even work outside unless declared outside?

Comment (joeybaker) [inline README.md]: yes, good point!

**Comment by cscheye** [comment]
Thanks for doing all that work @joeybaker!

File: README.md [hunk 3]
```diff
@@ -410,37 +466,131 @@ function onThingHappended(cb){
...
+    const lukeSkywalker = 'Luke Skywalker'
+
+    // bad
+    const obj = {
+      lukeSkywalker: lukeSkywalker
+    }
+
+    // good
+    const obj = {
+      lukeSkywalker
```
Comment (tedbreen) [inline README.md:545]: this is awkward, but I suppose i'll get use to it.

Comment (joeybaker) [inline README.md:545]: yea, it's pretty weird at first, but then you start using it, and it's really nice. 

``` js
// this sucks
return {firstName: firstName, lastName: lastName, age: age}
// this is much nicer
return {firstName, lastName, age}
```

File: README.md [hunk 4]
```diff
@@ -957,8 +1398,9 @@ function onThingHappended(cb){
     if (test && test.arg === doSomethingWith(5) && puppies() === 'cute') ciderHouse = true
 
     // good
-    if (test && test.arg === doSomethingWith(5) && puppies() === 'cute')
+    if (test && test.arg === doSomethingWith(5) && puppies() === 'cute'){
```
Comment (techwraith) [inline README.md]: Need a space here: `) {`

**Comment by techwraith** [comment]
Other than that last comment, this looks pretty good.

**Comment by tedbreen** [comment]
There's a lot to digest here, but it should be good. :+1:

**Comment by joeybaker** [comment]
Okay. Merging.

**Pull request merged**